src/brouwers/shop/payments/sisow/service.py

```diff
@@ -0,0 +1,91 @@
+from functools import lru_cache
+from typing import Iterator, List
+from urllib.parse import unquote
+
+from django.urls import reverse
+
+from ...models import Payment, ShopConfiguration
+from .api import NS, calculate_ideal_sha1, calculate_sha1, xml_request
+from .constants import Payments
+from .exceptions import InvalidIssuerURL
+
+
+class iDealBank:
+
+    def __init__(self, _id: str, name: str):
+        self.id = _id
+        self.name = name
+
+
+@lru_cache()
+def get_ideal_banks() -> List[iDealBank]:
+    root = xml_request('DirectoryRequest')
+    _issuers = root.findall("*/{{{ns}}}issuer".format(ns=NS))
+    banks = [
+        iDealBank(
+            _id=issuer.find("{{{ns}}}issuerid".format(ns=NS)).text,
+            name=issuer.find("{{{ns}}}issuername".format(ns=NS)).text,
+        )
+        for issuer in _issuers
+    ]
+    return banks
+
+
+def get_ideal_bank_choices() -> Iterator:
```

Comment:
Why is this a generator and not just a tuple/list?

Reply:
slighly more elegant, no other reason than that. there's not really a need for a `list` object here either